text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,155 @@
+  - `resource/` - Contents of `{specification}/resource/semantic_conventions`
+  - `trace/` - Contents of `{specification}/trace/semantic_conventions`
+  - `metrics/` - Contents of `{specification}/metrics/semantic_conventions`
+  - `logs/`- Contents of `{specification}/logs/semantic_conventions`
+  - `schemas/` - A new location for [Telemetry Schemas](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/README.md)
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/semantic_conventions/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+deprecated, with documentation denoting the move.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+director structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic convetions used to implement API/SDK details will be fully specified
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location.
+
+- A new repository `opentelemetry/semantic_conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- All semantic conventions in the Specification will be copied to the new
+  repository.
+- Semantic conventions in the Specification will be marked as deprecated with
+  links to the new location.
+- The Specification will be updated to require "special" conventions, like
+  `service.name` and configuration interaction.
+- Instrumentation authors will update their `SchemaURL` to
+   `https://opentelemetry.io/semantic_conventions/schemas/{semconv_version}`
+   from
+   `https://opentelemetry.io/schemas/{spec_version}`
```

Comment:
Is this effort/pain worth it or should we just keep the original URL form?

Reply:
Or would this only apply to users updating to new versions post this move?

Reply:
This is because there will be a version split from the specification.  So the pain of moving is ONLY a  onetime thing for existing users, but also the new repository can start at a non-1.0 version number to denote the current instability.

Reply:
> the new repository can start at a non-1.0 version number to denote the current instability

If we intend to indicate that using the single global version number, we'd never be able to go semconv `v1.0` though as there will always be some (new) unstable convention. (A subset of) HTTP that's considered stable would still be part of the `0.x` conventions. Maybe we should just go with `1.0.0` right away and (based on your "what stability actually enforces" proposal and other wording) define how to interpret versioning in the new repo. I assume we're keeping the stable/unstable markers for documents and sections to distinguish stability within the overall semantic-conventions repo.

Reply:
Maybe we should not use semver for the repository as a whole, but version conventions, or sets thereof, independently?

Reply:
> an entire InstrumentationScope (Meter, Tracer, Logger) MUST be the same schema version family

This is problematic anyways, at least for tracing, because e.g. an exception event might be created by a different instrumentation using a different semconv version than the main span.

Reply:
How would we approach HTTP stability wrt versions? Sounds like we'll need to split schema versions almost right away. 

Perhaps we can solve it with two versions:
- the default repo-level version set to 0.X applied to all unstable conventions
- the 'stable' version (1.X) applied to all stable conventions

Reply:
@lmolkova How would that work in relation to SchemaURL concerns I raised above?  I.e. is it technically feasible for us to have two versions split in one repo?

Reply:
>This is because there will be a version split from the specification.

Why this is necessary? Why can't we continue version numbers 1.20+ in the new repo?

Reply:
> How would that work in relation to SchemaURL concerns I raised above? I.e. is it technically feasible for us to have two versions split in one repo?

@jsuereth I can experiment to see if something goes wrong. On the high-level I don't see an issue if we think about version for the whole convention and not individual parts:
- Tracer/meter created for HTTP instrumentation sets schema to 1.X.  network conventions are shaded in and cannot move on their own.
- Tracer/meter for pure socket communication, gRPC, DB, etc set url to schema 0.X

Reply:
I have a feeling we are overcomplicating things. What exactly is the problem we are solving by giving different parts of semconv different version numbers?

Reply:
We also want to ship stable instrumentation libraries for HTTP,  so we need to have different [semconv artifacts](https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-semconv) generated (stable and alpha).

So we have to have a way to distinguish stable and unstable conventions.

We can do it by separating semconv to `stable` and `experimental` folders and making tooling understand it.
Or by adding a `version` property in yaml that applies to the group.
E.g. 

```diff
  - id: network-core
    prefix: net
    type: attribute_group
    brief: These attributes may be used for any network-related operation.
+ version: 1.0.0

or

+ stable: true
    ...
````

In any case, this is needed regardless of where conventions live, @trask I hope we can discuss it with HTTP crew.

Reply:
>  What exactly is the problem we are solving by giving different parts of semconv different version numbers?

@tigrannajaryan from my point of view, we can have the same version number, but the stable/experimental distinction is essential to ship instrumentation libraries with corresponding stability guarantees.

Reply:
Totally agree with the need to distinguish stable and unstable conventions. I am just saying I don't think we should use version numbers to be the designators of the stability.

Reply:
@tigrannajaryan I agree we can avoid using version numbers to denote stability of conventions.

However, we *do* need something denote when stability has broken.  Without being able to leverage major version numbers, I'm concerned we're stuck.

Reply:
>However, we do need something denote when stability has broken. Without being able to leverage major version numbers, I'm concerned we're stuck.

@jsuereth OK, sounds good. I am still not sure I understand _why_ we need denote the stability. For whom is this information? Is it for schema file consumers? Is it for semconv creators? Is it for instrumentation developers? Depending on the answer we may want a different solution.

Reply:
WE talked about this offline.  We want to denote stability in two fashions:

1. Programatically via SchemaURL
2. Intrinsically via Semantic Versioning of the SchemaURL version.

We expect (2) to catch the attention of OTEL Users who don't pay attention to details/nuances of our designs + discussions.